src/graph/PermissionCheck.cpp

```diff
@@ -0,0 +1,139 @@
+             */
+            return true;
+        }
+        case Sentence::Kind::kCreateSpace :
+        case Sentence::Kind::kDropSpace :
+        case Sentence::Kind::kCreateSnapshot :
+        case Sentence::Kind::kDropSnapshot :
+        case Sentence::Kind::kBalance :
+        case Sentence::Kind::kAdmin :
+        case Sentence::Kind::kConfig :
+        case Sentence::Kind::kIngest :
+        case Sentence::Kind::kDownload : {
+            return permission::PermissionManager::canWriteSpace(session);
+        }
+        case Sentence::Kind::kCreateTag :
+        case Sentence::Kind::kAlterTag :
+        case Sentence::Kind::kCreateEdge :
+        case Sentence::Kind::kAlterEdge :
+        case Sentence::Kind::kDropTag :
+        case Sentence::Kind::kDropEdge :
+        case Sentence::Kind::kCreateTagIndex :
+        case Sentence::Kind::kCreateEdgeIndex :
+        case Sentence::Kind::kDropTagIndex :
+        case Sentence::Kind::kDropEdgeIndex : {
+            return permission::PermissionManager::canWriteSchema(session);
+        }
+        case Sentence::Kind::kCreateUser :
+        case Sentence::Kind::kDropUser :
+        case Sentence::Kind::kAlterUser : {
+            return permission::PermissionManager::canWriteUser(session);
+        }
+        case Sentence::Kind::kRevoke :
+        case Sentence::Kind::kGrant : {
+            /**
+             * Use grant and revoke are special operations.
+             * Because have not found the target space id and target role
+             * so permission checking needs to be done in their executor.
+             * skip the check at here.
+             */
+            return true;
+        }
+        case Sentence::Kind::kRebuildTagIndex :
+        case Sentence::Kind::kRebuildEdgeIndex :
+        case Sentence::Kind::kInsertVertex :
+        case Sentence::Kind::kUpdateVertex :
+        case Sentence::Kind::kInsertEdge :
+        case Sentence::Kind::kUpdateEdge :
+        case Sentence::Kind::kDeleteVertex :
+        case Sentence::Kind::kDeleteEdges : {
+            return permission::PermissionManager::canWriteData(session);
+        }
+        case Sentence::Kind::kDescribeTag :
+        case Sentence::Kind::kDescribeEdge :
+        case Sentence::Kind::kDescribeTagIndex :
+        case Sentence::Kind::kDescribeEdgeIndex :
+        case Sentence::Kind::kGo :
+        case Sentence::Kind::kSet :
+        case Sentence::Kind::kPipe :
+        case Sentence::Kind::kMatch :
+        case Sentence::Kind::kAssignment :
+        case Sentence::Kind::kLookup :
+        case Sentence::Kind::kYield :
+        case Sentence::Kind::kOrderBy :
+        case Sentence::Kind::kFetchVertices :
+        case Sentence::Kind::kFetchEdges :
+        case Sentence::Kind::kFindPath :
+        case Sentence::Kind::kLimit :
+        case Sentence::Kind::KGroupBy :
+        case Sentence::Kind::kReturn : {
+            return permission::PermissionManager::canReadSchemaData(session);
```

Comment:
canReadData?

Reply:
> Maybe canReadSchemaOrData is a better name.

Good idea. Changed method name from `canReadSchemaData` to `canReadSchemaOrData`